File: tests/human/x/php/group_by_conditional_sum.php
<?php
$items = [
    ["cat" => "a", "val" => 10, "flag" => true],
    ["cat" => "a", "val" => 5, "flag" => false],
    ["cat" => "b", "val" => 20, "flag" => true],
];
$groups = [];
foreach ($items as $it) {
    $groups[$it['cat']][] = $it;
}
ksort($groups);
$result = [];
foreach ($groups as $cat => $group) {
    $total = 0;
    $flagged = 0;
    foreach ($group as $x) {
        $total += $x['val'];
        if ($x['flag']) {
            $flagged += $x['val'];
        }
    }
    $result[] = ["cat" => $cat, "share" => $flagged / $total];
}
var_dump($result);
?>

File: tests/human/x/python/group_by_conditional_sum.py
items = [
    {"cat": "a", "val": 10, "flag": True},
    {"cat": "a", "val": 5, "flag": False},
    {"cat": "b", "val": 20, "flag": True},
]

# Group by category and compute conditional share
stats = {}
for itm in items:
    g = stats.setdefault(itm["cat"], {"num": 0, "den": 0})
    if itm["flag"]:
        g["num"] += itm["val"]
    g["den"] += itm["val"]

result = [
    {"cat": cat, "share": (g["num"] / g["den"] if g["den"] else 0)}
    for cat, g in stats.items()
]
result.sort(key=lambda r: r["cat"])
print(result)
